osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -490,5 +497,72 @@ QueryData getTemperatures(QueryContext &context) {
 
   return results;
 }
+
+QueryData getFanSpeeds(QueryContext &context) {
+  QueryData results;
+
+  SMCHelper smc;
+  if (!smc.open()) {
+    return {};
+  }
+
+  // Get number of fans.
+  QueryData key_data;
+  genSMCKey("FNum", smc, key_data);
+
+  if (key_data.empty()) {
+    // The SMC search for key information failed.
+    return results;
+  }
+
+  auto &smcRow = key_data.back();
+  if (smcRow["value"].empty()) {
+    return results;
+  }
+
+  // Get attributes for each fan.
+  int numFans = std::stoi(smcRow["value"]);
+  for (int fanIdx = 0; fanIdx < numFans; fanIdx++) {
+    Row r;
+    r["fan"] = std::to_string(fanIdx);
+
+    for (const auto &smcFanSpeedKey : kSMCFanSpeeds) {
+      std::stringstream key;
+      key << boost::format(smcFanSpeedKey.first) % fanIdx;
+
+      QueryData key_data;
+
+      genSMCKey(key.str(), smc, key_data);
+      if (!key_data.empty()) {
+        auto &smcRow = key_data.back();
+        if (!smcRow["value"].empty()) {
+          std::string val;
+          try {
+            val = boost::algorithm::unhex(smcRow["value"]);
+          } catch (const boost::algorithm::hex_decode_error &e) {
+            continue;
+          }
+
+          if (val.size() != 2) {
+            continue;
+          }
+
+          int fanSpeed = (int(val[0]) << 6) + (int(val[1]) >> 2);
+          r[smcFanSpeedKey.second] = std::to_string(fanSpeed);
```

Comment:
We have a helper macro `INTEGER()` that's preferred over `std::to_string` *only* when assigning column values. The macro will help if/when we convert `Row`s to dynamics and can support the literal types via a union or something smart. ;)